File: atlhyper_master_v2/model/resource_quota.go
// atlhyper_master_v2/model/resource_quota.go
// ResourceQuota Web API 响应类型（camelCase JSON tag，扁平结构）
package model

// ResourceQuotaItem ResourceQuota 列表项
type ResourceQuotaItem struct {
	Name      string            `json:"name"`
	Namespace string            `json:"namespace"`
	Scopes    []string          `json:"scopes,omitempty"`
	Hard      map[string]string `json:"hard,omitempty"`
	Used      map[string]string `json:"used,omitempty"`
	CreatedAt string            `json:"createdAt"`
	Age       string            `json:"age"`
}

// ResourceQuotaDetail ResourceQuota 详情
type ResourceQuotaDetail struct {
	Name      string            `json:"name"`
	Namespace string            `json:"namespace"`
	Scopes    []string          `json:"scopes,omitempty"`
	Hard      map[string]string `json:"hard,omitempty"`
	Used      map[string]string `json:"used,omitempty"`
	CreatedAt string            `json:"createdAt"`
	Age       string            `json:"age"`

	Labels      map[string]string `json:"labels,omitempty"`
	Annotations map[string]string `json:"annotations,omitempty"`
}

File: atlhyper_web/src/api/resource-quota.ts
/**
 * ResourceQuota API
 *
 * 后端已完成 model_v2 → model 扁平化转换，前端直接使用
 */

import { get } from "./request";

// ============================================================
// 类型定义（匹配后端 model 响应）
// ============================================================

export interface ResourceQuotaItem {
  name: string;
  namespace: string;
  scopes?: string[];
  hard: Record<string, string>;
  used: Record<string, string>;
  createdAt: string;
  age: string;
}

export interface ResourceQuotaDetail {
  name: string;
  namespace: string;
  scopes?: string[];
  hard: Record<string, string>;
  used: Record<string, string>;
  createdAt: string;
  age: string;
  labels?: Record<string, string>;
  annotations?: Record<string, string>;
}

// ============================================================
// 响应类型（内部使用）
// ============================================================

interface ListResponse<T> {
  message: string;
  data: T[];
  total: number;
}

interface DetailResponse<T> {
  message: string;
  data: T;
}

// ============================================================
// API 查询参数（内部使用）
// ============================================================

interface ClusterResourceParams {
  cluster_id: string;
  namespace?: string;
}

// ============================================================
// API Functions
// ============================================================

export function getResourceQuotaList(params: ClusterResourceParams) {
  return get<ListResponse<ResourceQuotaItem>>("/api/v2/resource-quotas", params);
}

export function getResourceQuotaDetail(params: { ClusterID: string; Namespace: string; Name: string }) {
  return get<DetailResponse<ResourceQuotaDetail>>(
    `/api/v2/resource-quotas/${encodeURIComponent(params.Name)}`,
    { cluster_id: params.ClusterID, namespace: params.Namespace }
  );
}
